can/interfaces/socketcan/socketcan.py

```diff
@@ -127,18 +162,31 @@ def build_bcm_header(opcode, flags, count, ival1_seconds, ival1_usec, ival2_seco
     #     struct timeval ival1, ival2; ->  llll ...
     #     canid_t can_id; -> I
     #     __u32 nframes; -> I
-    bcm_cmd_msg_fmt = "@3I4l2I0q"
-
-    return struct.pack(bcm_cmd_msg_fmt,
-                       opcode,
-                       flags,
-                       count,
-                       ival1_seconds,
-                       ival1_usec,
-                       ival2_seconds,
-                       ival2_usec,
-                       can_id,
-                       nframes)
+
+    fields = [
+        ("opcode", ctypes.c_uint32),
+        ("flags", ctypes.c_uint32),
+        ("count", ctypes.c_uint32),
+        ("ival1_tv_sec", ctypes.c_long),
+        ("ival1_tv_usec", ctypes.c_long),
+        ("ival2_tv_sec", ctypes.c_long),
+        ("ival2_tv_usec", ctypes.c_long),
+        ("can_id", ctypes.c_uint32),
+        ("nframes", ctypes.c_uint32),
+    ]
+    BcmMsgHead = bcm_header_factory(fields)
```

Comment:
Consider calling the factory once at import time or socketcan bus instantiation time.

Reply:
Is there a specific test file that I should put it under? Not sure if I just missed one for the socketcan interface

Reply:
Done